tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -371,7 +372,7 @@ static PyObject * THPVariable_index_scalar(PyObject* self, PyObject* args) {
   // TODO: change the condition to `self_.dim() != 0` once we expose scalars
   // in PyTorch.
   if (!isIntegralType(self_.scalar_type(), /*includeBool=*/true) || self_.sym_numel() != 1) {
-    throw TypeError("only integer tensors of a single element can be converted to an index");
+    C10_THROW_ERROR(TypeError, "only integer tensors of a single element can be converted to an index");
```

Comment:
Is there a difference between this and `TORCH_CHECK_TYPE(false, ...)`? You do use both in this PR.

Reply:
C10_THROW_ERROR can't format string, otherwise they are the same. Because the error message is a single string here, there is no difference.